src/components/admin/jokes/AdminJokesList.vue

```diff
@@ -2,164 +2,188 @@
-
-                let start = current - Math.floor(length / 2)
-                start = Math.max(start, min)
-                start = Math.min(start, min + total - length)
-                
-                return Array.from({length: length}, (el, i) => start + i)
-            },
-            changeCategory(category){
-                this.currentCategory = category
-                if(category !== this.$t('adminJokes.allCategories')){
-                    if(this.categories.indexOf(this.$t('adminJokes.allCategories')) === -1)
-                        this.categories.push(this.$t('adminJokes.allCategories'))
-                    this.data = this.jokeAdmin.filter((e) => {
-                        if(e.category === category) return true
-                        else return false
-                    })
-                    this.total = this.data.length
-                    this.totalPages = Math.ceil(this.total/this.perPage)
-                    this.currentPage = 1
-                }
-                else{
-                    this.data = this.jokeAdmin
-                    this.total = this.jokeAdmin.length
-                    this.totalPages = Math.ceil(this.total/this.perPage)
+        editJoke(jokeId){
+            this.jokes.map(value => {
+                if(value.id === jokeId){
+                    this.editJokeTmp.orginalJoke = value
+                    this.editJokeTmp.modifiedJoke = value.content
                 }
-                
-            }
+            })
+            this.$bvModal.show("edit-joke-modal");  
         },
-        computed: {
-            ...mapGetters(['jokeAdmin']),
-            paginatedData(){
-                let start = (this.currentPage - 1) * this.perPage, end = start + this.perPage
-                return this.data.slice(start, end)
-            },
-            pages() {
-                let check = this.getPagingRange(this.currentPage, {min: 1, total: this.totalPages, length: this.maxVisibleButtons})
-                let range = []
-                for (let i = 0; i <check.length; i+= 1 ) {
-                    range.push({
-                    number: check[i],
-                    isDisabled: check[i] === this.currentPage 
-                    });
+        modifyJoke(e){
+            this.editJokeTmp.errors = []
+            let errors = service.validateEditJoke({content: this.editJokeTmp.modifiedJoke})
+            if(errors.length > 0)
+                this.editJokeTmp.errors = errors
+            else{
+                let joke = {
+                    id: this.editJokeTmp.orginalJoke.id,
+                    joke: this.editJokeTmp.modifiedJoke
                 }
-                return range
-            },
-            isInFirstPage() { 
-                return this.currentPage === 1 
-            },
-            isInLastPage() { 
-                return this.currentPage === this.totalPages
-            },
+                service.modifyJoke(joke).then(response => {
```

Comment:
Tutaj też odwróćmy ify, e.preventDefault() na pewno potrzebny? Vue powinno to ogarnąć z defaulta:)